osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -293,7 +293,7 @@ protected sealed override void RevertResultInternal(JudgementResult result)
 
         protected virtual double GetBonusScoreChange(JudgementResult result) => Judgement.ToNumericResult(result.Type);
 
-        protected virtual double GetComboScoreChange(JudgementResult result) => Judgement.ToNumericResult(result.Type) * (1 + result.ComboAfterJudgement / 10d);
+        protected virtual double GetComboScoreChange(JudgementResult result) => Judgement.ToNumericResult(result.Type) * Math.Pow(result.ComboAfterJudgement, 0.5);
```

Comment:
both of these functions get overridden by the ruleset-specific processors, except for `GetComboScoreChange` in `OsuScoreProcessor`. is this intended for just osu! or all rulesets?

Reply:
Good catch: I didn't know ComputeTotalScore was overridden in OsuScoreProcessor, so mybad.
The changes to ComputeTotalScore where affecting the Scoring test scene though, which is why I assumed it wasn't overridden but this is because apparently that scene is using the raw ScoreProcessor. (guess I'll fix that in another PR)

The PR is still a draft. The changes was originally intended to be for osu! only but potentially benefit other gamemodes too, but as I gave a deeper look at the code of other game-modes, they already have an alternative solution (logarithm of combo) and require different accuracy exponents. Thus I believe they do not need this change but it can be a good idea to ask experts from those rulesets if they think this new alternative can also work better in their ruleset.

One question this brings up is: does it make sense to override ComputeTotalScore/GetComboScoreChange in OsuScoreProcessor ? If yes, then what should be the default implementation in ScoreProcessor since all others rulesets already override it ? Wouldn't it make more sense to have ScoreProcessor be abstract and thus ComputeTotalScore and GetComboScoreChange too ?

Reply:
The intention for it not being abstract is to make it easier for third party ruleset developers to get to a running state without having to worry about how score is implemented. So, in a similar vein, the implementation there(/here) should be something that leads to good values in a general case.  
For example, ask the question of: if the taiko implementation were to use this implementation - would it behave better with the previous version or this new one?

`OsuScoreProcessor` is a bit of a weird one. I wouldn't be against removing the overrides there.

Reply:
I just noticed you answer here.
The taiko, mania and catch scoring would definitely be better with the square-root combos than raw combo as the former is closer to their current scoring implementation. (sqrt(combo) is closer to log(combo) than just raw combo)
But based on your argument, it might make more sense to have log(combo) as the default since all other game-modes use the logarithm ?

Reply:
Honestly, if we're just looking to get things moving, then the base implementation should not be changed (and your change should be moved to `OsuScoreProcessor` as to not break other rulesets).

But that said, I don't know how much this matters in such an early state. At worst it's going to cause differences in local user scores for users playing non-official rulesets.

@smoogipoo thoughts?

Reply:
On second thoughts maybe it's just fine to make this a breaking change..

Reply:
I asked above: "would [taiko] behave better with the previous version or this new one?".

The response I was "the [...] scoring would definitely be better with the square-root combos".

Based on that I'm inclined to believe the new method would be better as a base implementation, if all of our rulesets would benefit from this.

Reply:
Yes. Taiko, mania and ctb override this anyway, but square-root combos is closer to what those 3 rulesets already use (which is log of combo + some hard-cap), thus it seems safe to assume that this is a better starting base for future rulesets.

What might make sense is to have taiko as the default though, and `osu!` overriding it, since I think mania and ctb are closer to taiko's combo scaling. I suppose it's not worth spending too much time on that decision though ?